fabric-interaction-events-v1/src/main/java/net/fabricmc/fabric/impl/interaction/InteractionRaycasting.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.interaction;
+
+import java.util.Objects;
+
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.GameMode;
+
+/**
+ * Utilities related to raycasting in preparation for calling interaction events.
+ */
+public final class InteractionRaycasting {
+	/**
+	 * Gets the reach distance of a player.
+	 *
+	 * @param player the player
+	 * @return the reach distance
+	 */
+	public static double getReachDistance(ServerPlayerEntity player) {
```

Comment:
Why `ServerPlayerEntity` and not just `PlayerEntity`?

Reply:
I don't believe that will be needed on the logical client, but time will tell there